### DISCUSSION TITLE
Option to Remove Steam Verification when Purchasing Paid Shop Packs

### DISCUSSION THREAD

**Discussion by Decollates** [gameserverapp/Platform#371] [Suggestions]
Pretty much what it says on the title. My community has branched out to another game that does not utilize steam as its launching platform, and in the effort to provide options for people who want to support the servers we host for them, I'm attempting to set up paid shop packs for them. But if there's a way that the packs themselves can "opt out" of requiring a steam OAuth to validate them when they purchase, that would be immensely beneficial for our plans.

**Comment by mvaessen** [comment]
Hi,

Thanks for your suggestion!

I'm not entirely sure I understand the scope of your suggestion. We do need some form of authentication to know who bought something. Is your suggestion about adding a user+pass login instead of steam?

**Comment by Decollates** [comment]
Not necessarily. We can simplify it to being "discord exclusive" purchases, as in purchases that only affect users for Discord. If that can entail setting up the pack so that it uses uses a Discord OAuth instead of a Steam one, would that work?

**Comment by mvaessen** [comment]
Theoretically that is possible for sure.
Options like sending ingame commands / whitelists etc wont be available in those cases though. Basically just discord actions.

Since people already have their discord account connected to their steam account on the community website, we could transform the login system into a multi-social login. That would allow people to optionally log into their main account using any of their connected social accounts.

Would also be convenient for when players decide to join a different server on your community that only supports steam.

**Comment by Decollates** [comment]
Yeah, at that point it's ok if it's discord only. This particular game we're hosting is not available through steam, but its own launcher, so my team is having to do some workarounds - GSA's potential already makes things incredibly easy to manage with discord-exclusive stuff. I do worry about how disruptive it might be for the rest of GSA's workings if steam logins are removed as the default? I'm not sure; that's why I suggested the bypass option, unless you come up with something better.

**Comment by Decollates** [comment]
I noticed that shop packs/supporter tiers can utilize different games as the basis for their commands. If so, how does it work for a community that is exclusively Minecraft, since Minecraft doesn't go through Steam? 
![image](https://user-images.githubusercontent.com/44717221/226150006-5188e500-6884-4561-9fff-9f5807fb9583.png)

**Comment by Decollates** [comment]
As an alternative -- instead of a whole new account creation system, can we have the option to use either Steam oAuth (default), or Discord's auth?

**Comment by mvaessen** [comment]
rejecting since this is soon no longer applicable.
